lib/carrot/amqp/server.rb

```diff
@@ -125,7 +126,7 @@ def socket
 
       begin
         # Attempt to connect.
-        mutex.lock if multithread?
+        @mutex.lock if multithread?
```

Comment:
@olleolleolle you could probably just use `@mutex&.lock`/`@mutex&.unlock`?

Reply:
That would be less work for the computer! So, great!